desktop_version/src/Music.cpp

```diff
@@ -13,165 +14,546 @@
+        format.nSamplesPerSec = spec.freq;
+        format.wFormatTag = FAUDIO_FORMAT_PCM;
+        format.wBitsPerSample = 16;
+        format.nBlockAlign = format.nChannels * format.wBitsPerSample;
+        format.nAvgBytesPerSec = format.nSamplesPerSec * format.nBlockAlign;
+        format.cbSize = 0;
     }
 
     void Dispose()
     {
-        /* Destroy all track voices, SDL_free buffer from LoadWAV */
-        Mix_FreeChunk(m_sound);
+        SDL_free(wav_buffer);
     }
 
     void Play()
     {
-        /* Fire-and-forget from a per-track FAudioSourceVoice pool */
-        if (Mix_PlayChannel(-1, m_sound, 0) == -1)
+        for (int i = 0; i < VVV_MAX_CHANNELS; i++)
         {
-            vlog_error("Unable to play WAV file: %s", Mix_GetError());
+            FAudioVoiceState voicestate;
+            FAudioSourceVoice_GetState(voices[i], &voicestate, 0);
+            if (voicestate.BuffersQueued == 0)
+            {
+                FAudioVoiceDetails details;
+                FAudioVoice_GetVoiceDetails(voices[i], &details);
+                if (details.InputChannels != format.nChannels)
+                {
+                    FAudioVoice_DestroyVoice(voices[i]);
+                    FAudio_CreateSourceVoice(faudioctx, &voices[i], &format, 0, 2.0f, NULL, NULL, NULL);
+                }
+                const FAudioBuffer faudio_buffer = {
+                    FAUDIO_END_OF_STREAM, /* Flags */
+                    wav_length * 8, /* AudioBytes */
+                    wav_buffer, /* AudioData */
+                    0, /* playbegin */
+                    0, /* playlength */
+                    0, /* LoopBegin */
+                    0, /* LoopLength */
+                    0, /* LoopCount */
+                    NULL
+                };
+                if (FAudioSourceVoice_SubmitSourceBuffer(voices[i], &faudio_buffer, NULL))
+                {
+                    vlog_error("Unable to queue sound buffer");
+                    return;
+                }
+                if (FAudioSourceVoice_Start(voices[i], 0, FAUDIO_COMMIT_NOW))
+                {
+                    vlog_error("Unable to start voice processing");
+                }
+                return;
+            }
         }
     }
 
     static void Init(int audio_rate, int audio_channels)
     {
-        const Uint16 audio_format = AUDIO_S16SYS;
-        const int audio_buffers = 1024;
-
-        /* FAudioCreate, FAudio_CreateMasteringVoice */
-        if (Mix_OpenAudio(audio_rate, audio_format, audio_channels, audio_buffers) != 0)
+        if (FAudioCreate(&faudioctx, 0, FAUDIO_DEFAULT_PROCESSOR))
         {
-            vlog_error("Unable to initialize audio: %s", Mix_GetError());
-            SDL_assert(0 && "Unable to initialize audio!");
+            vlog_error("Unable to initialize FAudio");
+            return;
+        }
+        if (FAudio_CreateMasteringVoice(faudioctx, &masteringvoice, audio_channels, audio_rate, 0, 0, NULL))
```

Comment:
Ditto as `FAudioCreate`, down to not seeing an `FAudio_DestroyMasteringVoice` in the FAudio source.

Reply:
Yep, this can happen at the same place as FAudio_Release.